src/zenml/cli/stack_components.py

```diff
@@ -235,22 +238,133 @@ def register_stack_component_command(
                 cli_utils.error(str(e))
                 return
 
-            from zenml.stack.stack_component_class_registry import (
-                StackComponentClassRegistry,
-            )
+        from zenml.stack.registry import flavor_registry
 
-            component_class = StackComponentClassRegistry.get_class(
-                component_type=component_type, component_flavor=flavor
+        zenml_flavors = flavor_registry.get_flavors_by_type(
+            component_type=component_type
+        )
+
+        try:
+            # Try to find if there are any custom flavor implementations
+            flavor_wrapper = Repository().zen_store.get_flavor_by_name_and_type(
+                flavor_name=flavor,
+                component_type=component_type,
             )
-            component = component_class(name=name, **parsed_args)
+
+            # If there is one, check whether the same flavor exists as a default
+            # flavor to give out a warning
+            if flavor in zenml_flavors:
+                cli_utils.warning(
+                    f"The flavor that you are currently using "
+                    f"'{flavor_wrapper.name}' is overwriting the a flavor with "
+                    f"the same which is provided by ZenML. This could lead "
+                    f"to an unexpected behavior and should be avoided if "
+                    f"possible."
+                )
```

Comment:
This warning doesn't belong here, in my humble opinion. It should instead be a check that is done when registering a new flavor. In other words, if we prevent a custom flavor from overwriting a built-in or integration flavor, this shouldn't happen here.

Reply:
Thanks! I just refactored it, it is a part of the `get_flavor` method of the `Repository`.